src/udev/net/link-config.c

```diff
@@ -841,8 +841,6 @@ static int link_apply_alternative_names(Link *link, sd_netlink **rtnl) {
                         }
                 }
 
-        if (link->new_name)
-                strv_remove(altnames, link->new_name);
```

Comment:
Hmm, with this PR, e.g.,
1. eth0 will get altnames, foo and bar,
2. eth0 will be renamed to foo,
3. foo will get an additional altname eth0, that's problematic.

The basic idea of this PR sounds OK for me. But please make `rtnl_set_link_name()` refuse ethX or wlanX.

Reply:
@yuwata I see now that the behavior you are describing was introduced by https://github.com/systemd/systemd/pull/16501/commits/434a34838034347f45fb9a47df55b1a36e5addfd, and that my patch causes the "`old_name`" logic to execute on the initial renaming of `ethX`.

If this is not the expected outcome, should the "`old_name`" logic in `rtnl_set_link_name()` just be removed (i.e., don't set the old name as an alternative name just because the new name is a current alternative name)? Or, if you want this logic to remain, would it be better to only set `old_name` as an alternative name if `old_name` is equal to one of `ID_NET_NAME_PATH`, `ID_NET_NAME_SLOT`, etc.? I think this would be sufficient to ensure that e.g. `eth0` does not get set as an alternative name.